src/udps.cpp

```diff
@@ -246,21 +223,60 @@ size_t Udp::deliver(size_t maxcalls)
         }
     }
 
-    // something was processed
+    // done
+    return result;
+}
+
+/**
+ *  Invoke callback handlers for buffered raw responses
+ *  @param   watcher   to keep track if the parent object remains valid
+ *  @param   maxcalls  the max number of callback handlers to invoke
+ *  @return  number of callback handlers invoked
+ */
+size_t Udps::deliver(size_t maxcalls)
+{
+    // result variable
+    size_t result = 0;
+
+    // we are going to make a call to userspace, so we keep monitoring if `this` is not destructed
+    Watcher watcher(this);
+
+    // deliver from all sockets
+    for (auto &socket : _sockets) result += socket.deliver(&watcher, maxcalls);
+
+    // return the number of buffered responses that were processed
     return result;
 }
 
 /**
  *  Send a query to a nameserver (+open the socket when needed)
+ *  @param  ip      IP address of the nameserver
+ *  @param  query   the query to send
+ *  @param  buffersize
+ *  @return inbound
+ */
+Inbound *Udps::send(const Ip &ip, const Query &query, int32_t buffersize)
+{
+    // choose a socket
```

Comment:
Change algorithm:
- If there are sockets with no subscribers: use that one + mark it as current
- If all the sockets already have subscribers: use the current